src/WebJobs.Extensions.ApiHub/ApiHubFile.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.Azure.ApiHub;
+using Microsoft.Azure.WebJobs.Extensions.ApiHub.Common;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Extensions.ApiHub
+{
+    class ApiHubFile : IFileStreamProvider
+    {
+        private readonly IFileItem _fileSource;
+
+        public string Path
+        {
+            get
+            {
+                return _fileSource.Path;
+            }
+        }
+
+        public ApiHubFile(IFolderItem rootFolder, string path)
+        {
+            _fileSource = rootFolder.CreateFileAsync(path, true).GetAwaiter().GetResult();
```

Comment:
I haven't seen how/where this type is used, but it seems strange to me that it is issuing what looks like an expensive (possibly remote) file operation from its constructor. Seems like that should be moved to an Async method.